plugins/config/parsers/auto_constructed_tables.cpp

```diff
@@ -153,18 +153,31 @@ Status ATCConfigParserPlugin::update(const std::string& source,
     std::string columns_value;
     columns_value.reserve(256);
 
+    // Always add the implicit path column
+    columns.push_back(
+        make_tuple(std::string("path"), TEXT_TYPE, ColumnOptions::DEFAULT));
+    columns_value += "path,";
+
     if (!params.HasMember("columns") || !params["columns"].IsArray()) {
       LOG(WARNING) << "ATC Table: " << table_name
                    << " is misconfigured (no columns)";
     }
 
     for (const auto& column : params["columns"].GetArray()) {
-      if (column.IsString()) {
-        columns.push_back(make_tuple(std::string(column.GetString()),
-                                     TEXT_TYPE,
-                                     ColumnOptions::DEFAULT));
-        columns_value += std::string(column.GetString()) + ",";
+      if (!column.IsString()) {
+        LOG(WARNING) << "ATC Table: " << table_name
+                     << " is misconfigured. (non-string columns)";
+        continue;
+      }
+
+      if (std::string(column.GetString()) == std::string("path")) {
+        LOG(WARNING) << "ATC Table: Includes `path`, which is implicit";
```

Comment:
small suggestion, it is not clear that this warning means the table is not going to work. I suggest `ATC Table: " << table_name << " is misconfigured. The configuration includes 'path', which is an reserved column."`

Also, (while you are here) do you mind auditing the other warning messages such that the language and punctuation is consistent?

Reply:
It's not a reserved column today. If the ATC query returns `path`,  [virtual_sqlite_table.cpp#L64-L68] emits a warning and does not use the implicit. (https://github.com/osquery/osquery/blob/master/osquery/sql/virtual_sqlite_table.cpp#L64-L68):
```
  if (r.count("path") > 0) {
    LOG(WARNING) << "Row contains a path key, refusing to overwrite";
  } else {
    r["path"] = sqlite_db.string();
  }
```

So, today, I think we have 3 possible states:
1. path not specific, implicit column, unusable
2. path specified, not in query. implicit column, works
3. path specified, defined by query, implicit is overridden.

Do you think we need to keep supporting case 3 there?

Reply:
I opted to keep supporting them. This leaves the underlying errors roughly the same:
1. If the config does nothing, you get a path column
2. If you config declares the column, you get a warning, and the implicit one.
3. If the config defines a query that returns path, you get a warning, and your defined column

I also took a very light pass at the messages here